tests/test_dimension.py

```diff
@@ -1058,7 +1057,11 @@ def test_no_fusion_simple(self):
 
         op = Operator(eqns)
 
-        exprs = FindNodes(Expression).visit(op._func_table['bf0'].root)
+        # Check code generation
+        trees = retrieve_iteration_tree(op)
+        assert len(trees) == 1
+        assert all(i.dim.is_Incr for i in trees[0][1:5])
```

Comment:
you're implementing the check:  "this `tree` is expected to be a blocked loop nest" almost always as something along the lines of `assert all(i.dim.is_Incr for i in trees[1][1:4])` . IMHO, you could add a utility function, say **for example** `assert_blocking(tree, index, depth)`, which runs systematically at least the following checks:

```
assert tree.root.dim.is_Time  
assert all(i.dim.is_Incr for i in tree[index:depth])
...
```

Reply:
Following as discussed